browser/model/installable-item.js

```diff
@@ -113,6 +116,38 @@ class InstallableItem {
     success();
   }
 
+  checkAndDownload(dl,url,sha,user,pass) {
```

Comment:
I believe we should provide unit test with new methods. Please add tests for this method. And if you could remove any unnecessary debug messages, that would be smashing.